src/System.Runtime.Extensions/tests/System/IO/Path.cs

```diff
@@ -215,104 +314,187 @@ public static void GetTempFileName()
     }
 
     [Fact]
-    public static void GetFullPath()
+    public static void GetFullPath_InvalidArgs()
     {
-        // Basic invalid arg checks
         Assert.Throws<ArgumentNullException>(() => Path.GetFullPath(null));
-        Assert.Throws<ArgumentException>(() => Path.GetFullPath(""));
-        Assert.Throws<ArgumentException>(() => Path.GetFullPath("http://www.microsoft.com"));
-        Assert.Throws<ArgumentException>(() => Path.GetFullPath("file://www.microsoft.com"));
+        Assert.Throws<ArgumentException>(() => Path.GetFullPath(string.Empty));
+    }
 
-        // Basic expansions (e.g. self to self, period to self, normalization of lots of periods, etc.)
+    [Fact]
+    public static void GetFullPath_BasicExpansions()
+    {
         string curDir = Directory.GetCurrentDirectory();
+
+        // Current directory => current directory
         Assert.Equal(curDir, Path.GetFullPath(curDir));
+
+        // "." => current directory
         Assert.Equal(curDir, Path.GetFullPath("."));
+
+        // ".." => up a directory
+        Assert.Equal(Path.GetDirectoryName(curDir), Path.GetFullPath(".."));
+
+        // "dir/./././." => "dir"
         Assert.Equal(curDir, Path.GetFullPath(Path.Combine(curDir, ".", ".", ".", ".", ".")));
-        Assert.Equal(curDir, Path.GetFullPath(curDir + Path.DirectorySeparatorChar + Path.DirectorySeparatorChar + Path.DirectorySeparatorChar + "."));
+
+        // "dir///." => "dir"
+        Assert.Equal(curDir, Path.GetFullPath(curDir + new string(Path.DirectorySeparatorChar, 3) + "."));
+
+        // "dir/../dir/./../dir" => "dir"
         Assert.Equal(curDir, Path.GetFullPath(Path.Combine(curDir, "..", Path.GetFileName(curDir), ".", "..", Path.GetFileName(curDir))));
+
+        // "C:\somedir\.." => "C:\"
         Assert.Equal(Path.GetPathRoot(curDir), Path.GetFullPath(Path.Combine(Path.GetPathRoot(curDir), "somedir", "..")));
+
+        // "C:\." => "C:\"
         Assert.Equal(Path.GetPathRoot(curDir), Path.GetFullPath(Path.Combine(Path.GetPathRoot(curDir), ".")));
 
-        // Try out a long path that normalizes down to less than MaxPath
+        // "C:\..\..\..\.." => "C:\"
+        Assert.Equal(Path.GetPathRoot(curDir), Path.GetFullPath(Path.Combine(Path.GetPathRoot(curDir), "..", "..", "..", "..")));
+
+        // "C:\\\" => "C:\"
+        Assert.Equal(Path.GetPathRoot(curDir), Path.GetFullPath(Path.GetPathRoot(curDir) + new string(Path.DirectorySeparatorChar, 3)));
+
+        // Path longer than MaxPath that normalizes down to less than MaxPath
         var longPath = new StringBuilder(curDir);
-        for (int i = 0; i < 1000; i++)
+        for (int i = 0; i < 10000; i++)
```

Comment:
This isn't necessarily big enough.  But we don't seem to expose MaxPath (probably because we've always considered it a const?), so this seems fine.